src/libopensc/card-iasecc.c

```diff
@@ -562,65 +552,29 @@ iasecc_init_amos(struct sc_card *card)
 	card->caps |= SC_CARD_CAP_APDU_EXT;
 	card->caps |= SC_CARD_CAP_USE_FCI_AC;
 
+	if (card->type == SC_CARD_TYPE_IASECC_MI)   {
+		resp_len = sizeof(resp);
+		rv = iasecc_select_aid(card, &MIIASECC_AID, resp, &resp_len);
+
+		rv = iasecc_mi_match(card);
```

Comment:
By convention, the match function should be used to determine the card's type. I suggest you add `iasecc_select_aid(card, &MIIASECC_AID, resp, &resp_len);` to `iasecc_mi_match()` (note that you should check `rv` when selecting the applet!). (Yes, I know that this has been a mistake of the past, but giving some love to the code today, will benefit all contributors in the future.)